core/src/main/java/org/mule/runtime/core/internal/connection/PoolingConnectionHandler.java

```diff
@@ -65,6 +65,9 @@ public void release() {
 
     boolean returnAttempted = false;
     try {
+      if (!connectionProvider.validate(connection).isValid()) {
+        return;
```

Comment:
add LOGGER.debug declaring that the connection was invalid and it will be destroyed instead.